frontend/src/test/java/com/google/tripmeout/frontend/servlet/ServletUtilTest.java

```diff
@@ -0,0 +1,152 @@
+package com.google.tripmeout.frontend.servlet;
+
+import static com.google.common.truth.Truth.assertThat;
+import static com.google.common.truth.Truth8.assertThat;
+import static org.junit.Assert.assertThrows;
+import static org.mockito.Mockito.when;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.servlet.http.HttpServletRequest;
+
+import com.google.gson.Gson;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.tripmeout.frontend.PlaceVisitModel;
+import com.google.tripmeout.frontend.TripModel;
+import com.google.tripmeout.frontend.serialization.GsonModelSerializationModule;
+import com.google.tripmeout.frontend.servlets.ServletUtil;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.junit.runners.JUnit4;
+import org.junit.runner.RunWith;
+
+@RunWith(JUnit4.class)
+public class ServletUtilTest {
+  @Mock HttpServletRequest request;
+  private Gson gson;
+
+  private final Pattern TRIP_URI_PATTERN = Pattern.compile(".*/trips/([^/]+)");
+  private final Pattern PLACE_URI_PATTERN = Pattern.compile(".*/trips/([^/]+)/places/([^/]+)");
+
+  @Before
+  public void setup() {
+    initMocks(this);
+    Injector injector = Guice.createInjector(new GsonModelSerializationModule());
+    this.gson = injector.getInstance(Gson.class);
+  }
+
+  @Test
+  public void extractFromRequestBody_supplyListClass_returnsOptionalList() throws IOException {
+    when(request.getReader())
+        .thenReturn(new BufferedReader(new StringReader("['hello', 'my', 'name', 'is']")));
+
+    Optional<List> extractedList = ServletUtil.extractFromRequestBody(request, gson, List.class);
```

Comment:
Called out above, but this is an `Optional<List<String>>` you want, so you probably want to use type tokens like:
```java
Optional<List<String>> extractedList = ServletUtil.extractFromRequestBody(request, gson, new TypeToken<List<String>>() {}.getType());
```

Which would use an overloaded version of the method using `java.lang.reflect.Type`